MekHQ/src/mekhq/campaign/stratcon/StratconContractDefinition.java

```diff
@@ -0,0 +1,420 @@
+package mekhq.campaign.stratcon;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.namespace.QName;
+import javax.xml.transform.Source;
+
+import mekhq.MekHQ;
+import mekhq.MekHqXmlUtil;
+
+/**
+ * This class holds data relevant to the various types of contract
+ * that can occur in the StratCon campaign system.
+ * @author NickAragua
+ *
+ */
+@XmlRootElement(name="StratconContractDefinition")
+public class StratconContractDefinition {
+    public static final String ROOT_XML_ELEMENT_NAME = "ScenarioTemplate";
+    public static final int COUNT_SCALED = -1;
+    
+    private static ContractDefinitionManifest definitionManifest;
+    private static Map<Integer, StratconContractDefinition> loadedDefinitions = new HashMap<>();
+    
+    static {
+        definitionManifest = ContractDefinitionManifest.Deserialize("./data/stratconcontractdefinitions/ContractDefinitionManifest.xml");
+        
+        // load user-specified modifier list
+        ContractDefinitionManifest userDefinitionList = ContractDefinitionManifest.Deserialize("./data/scenariomodifiers/UserContractDefinitionManifest.xml");
+        if(userDefinitionList != null) {
+            definitionManifest.definitionFileNames.putAll(userDefinitionList.definitionFileNames);
+        }
+    }
+    
+    /**
+     * Returns the stratcon contract definition for the given AtB Contract Type 
+     * as defined in AtBContract.java
+     */
+    public static StratconContractDefinition getContractDefinition(int atbContractType) {
+        definitionManifest = ContractDefinitionManifest.Deserialize("./data/stratconcontractdefinitions/ContractDefinitionManifest.xml");
+        loadedDefinitions.clear();
```

Comment:
This should likely be moved to an initialization method.

Also, it appears that it will _always_ reload the specific contract type because of that clear statement.